### PR TITLE
Document supported functionality in the README

### PR DESCRIPTION
**Pull request by SimonBarendse**
Make clear upfront what is and isn't yet supported in the README, and invite folks to contribute their desires by upvoting or filing issues.

I've changed the phrasing from "limitations" to a more roadmap-like todo list, to be more forward looking and bring across our intentions to continue expanding these SDKs across more use cases. I want it to be inviting to brainstorm or request additional use cases, and prevent these coming across as out of scope and SDKs not being suitable for these use cases.

Where there's existing issues already filed for functionality we don't support yet, I've included a link to the issue, so it's easier for folks to upvote this request if they need it too. We may even want to seed the issue tracker with issues for the other functionality, to make it even easier to indicate desire for these too. Before we do so, we may want to look into creating a central (core) repository first though, to track these, so that we're not duplicating issues for the same functionality across all repositories for the different languages, adding overhead to managing these issues and seeing demand across languages.

Where there's current alternatives available for programmatic access, I've listed these alternatives for easier discovery (e.g. Events API).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,17 +14,59 @@
...
+- [x] [Delete Items](https://developer.1password.com/docs/sdks/manage-items#delete-an-item)
+- [ ] List Items
+- [ ] Add & Update tags on items ([#86](https://github.com/1Password/onepassword-sdk-go/issues/86))
+
+Item/field Types:
+- [x] API Keys
+- [x] Passwords
+- [x] Logins
+- [x] Notes
+- [x] SSH Private Keys
```
Comment (Marton6) [inline README.md:36]: I don't think we should tick this as it's only supported in secret references. It's not supported in item create/get/update.

**Review by Marton6 — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (SimonBarendse) [reply README.md:36]:
```suggestion
- [x] SSH Private Keys (partially supported: supported in resolving secret references, not yet supported in item create/get/update)
```

Comment (SimonBarendse) [reply README.md:36]: @Marton6 I propose we document exactly that. So we achieve both:
1. Folks who just need to reach the private keys know they can achieve this through secret references, and don't unnecessarily get disappointed with not being able to do so and needing to explore alternative options.
2. Folks who need to create or edit SSH keys, or fetch the full item, know it's not yet possible.

Comment (SimonBarendse) [reply README.md]: Addressed in https://github.com/1Password/onepassword-sdk-go/pull/118/commits/53cfdf3498f48f4d2c0337d1586958fffd763aae. WDYT?

Comment (Marton6) [reply README.md]: I double checked how well SSH keys are supported in secret references and only retrieving the private key is supported. Other fields such as the public key, key type and fingerprint are not supported.

Comment (SimonBarendse) [reply README.md]: Thanks for checking!

**Review by libutcher — APPROVED** [review]
Approved! I think this will be super useful -- especially the ability to see what we've already filed issues for and click through and upvote features. I just left some small style edits and one larger suggestion.

After this is merged, do you want me to open matching updates to the Python and SDK repos?

File: README.md [hunk 2]
```diff
@@ -14,17 +14,59 @@
...
+- [x] Passwords
+- [x] Logins
+- [x] Notes
+- [x] SSH Private Keys (partially supported: supported in resolving secret references, not yet supported in item create/get/update)
+- [ ] SSH Public Keys
+- [ ] TOTP codes ([#93](https://github.com/1Password/onepassword-sdk-go/issues/93))
+- [ ] Files / Documents ([#108](https://github.com/1Password/onepassword-sdk-go/issues/108))
+- [x] URLs
+- [x] Credit card number & type
+- [x] Phone numbers
```
[the remark this answers is not in the document]
Comment (SimonBarendse) [reply README.md]: You're right, it's more accurate to talk about the supported fields. I've applied your suggestion.

File: README.md [hunk 3]
```diff
@@ -14,17 +14,59 @@
...
-
-1Password SDKs currently only support operations on text and concealed fields. As a result, you can't edit items that include information saved in other types of fields.
-
-When managing items with 1Password SDKs, you must use [unique identifiers (IDs)](https://developer.1password.com/docs/sdks/concepts#unique-identifiers) in place of vault, item, and field names.
+## Supported Functionality
+
+1Password SDKs are in active development. We're keen to hear what you'd like to see next. Let us know by [upvoting](https://github.com/1Password/onepassword-sdk-go/issues) or [filing](https://github.com/1Password/onepassword-sdk-go/issues/new/choose) an issue.
+
+### Item Management
+Operations:
```
[the remark this answers is not in the document]
Comment (SimonBarendse) [reply README.md:22]: Thank you

**Comment by SimonBarendse** [comment]
Thanks Lucy, this looks even better now! 😍  Agree that talking about the supported fields specifically is more accurate than the item templates these fields exist on, updated to reflect that.

> After this is merged, do you want me to open matching updates to the Python and SDK repos?

Sure, thanks for offering! One caveat for now to consider when creating these, is as I mentioned in the PR description, we currently track issues for adding functionality separately in each languages repo, so for now we'll need to check for which issues exist separately on those repos to link the right tickets instead of the ones we link here in the Go repo.

**Pull request merged**